### ISSUE TITLE
primary key in the update index method should be optionnal

### ISSUE THREAD

**Issue by bidoubiwa** [meilisearch/meilisearch-rust#296]
**Description**

Currently the method to update an index requires a primary key as per the method definition: 
```rust
pub async fn update(&self, primary_key: impl AsRef<str>) -> Result<(), Error>
```

As an example:

```rust
client.index("movies").update("movie_review_id"));
```

**Expected behavior**
In Meilisearch the update route accepts no primary Key field being defined, or one set to null. [See spec](https://specs.meilisearch.com/specifications/text/0132-indexes-api.html#_3-2-4-put-indexes-index-uid).

**Current behavior**
The `primaryKey` is required. It should not and could introduce a big breaking change if a new parameter becomes updatable on the update index route. In which case the user is going to have to define a primary key even if they only want to update the other parameter(s).

**Environment (please complete the following information):**

- Meilisearch version: v0.27.1

**Comment by phdavis1027** [comment]
I'll mess around with this!

**Comment by bidoubiwa** [comment]
Fixed by #315
